test_goto.py

```diff
@@ -71,63 +71,87 @@ def func():
 
-if sys.version_info >= (3, 6):
-    def test_jump_out_of_nested_2_loops():
-        @with_goto
-        def func():
-            x = 1
-            for i in range(2):
-                for j in range(2):
-                    # These are more than 256 bytes of bytecode, requiring
-                    # a JUMP_FORWARD below on Python 3.6+, since the absolute
-                    # address would be too large, after leaving two blocks.
-                    x += x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x
-                    x += x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x
-                    x += x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x
+def test_jump_out_of_nested_2_loops():
+    @with_goto
+    def func():
+        x = 1
+        for i in range(2):
+            for j in range(2):
+                # These are more than 256 bytes of bytecode
+                x += x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x
+                x += x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x
+                x += x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x+x
+
+                goto .end
+        label .end
+        return (i, j)
 
+    assert func() == (0, 0)
+
+def test_jump_out_of_nested_3_loops():
+    @with_goto
+    def func():
+        for i in range(2):
+            for j in range(2):
+                for k in range(2):
                     goto .end
-            label .end
-            return (i, j)
+        label .end
+        return (i, j, k)
 
-        assert func() == (0, 0)
+    assert func() == (0, 0, 0)
 
-    def test_jump_out_of_nested_3_loops():
-        def func():
-            for i in range(2):
-                for j in range(2):
-                    for k in range(2):
+def test_jump_out_of_nested_4_loops():
+    @with_goto
+    def func():
+        for i in range(2):
+            for j in range(2):
+                for k in range(2):
+                    for m in range(2):
                         goto .end
-            label .end
-            return (i, j, k)
-
-        pytest.raises(SyntaxError, with_goto, func)
-else:
-    def test_jump_out_of_nested_4_loops():
-        @with_goto
-        def func():
-            for i in range(2):
-                for j in range(2):
-                    for k in range(2):
-                        for m in range(2):
-                            goto .end
-            label .end
-            return (i, j, k, m)
+        label .end
+        return (i, j, k, m)
 
-        assert func() == (0, 0, 0, 0)
+    assert func() == (0, 0, 0, 0)
 
-    def test_jump_out_of_nested_5_loops():
-        def func():
-            for i in range(2):
-                for j in range(2):
-                    for k in range(2):
-                        for m in range(2):
-                            for n in range(2):
-                                goto .end
-            label .end
-            return (i, j, k, m, n)
+def test_jump_out_of_nested_5_loops():
```

Comment:
Aren't the `test_jump_out_of_nested_[2-5]_loops` tests redundant now, where we enable jumps out of arbitrary deep loops?

Reply:
I'll remove 3-5 as they indeed seem redundant, but will keep 2, as it's a good test in its own right, I think. 
(You can never have too many tests, e.g. if you've seen these somewhat-questionably-named "_survive/_live" test variations in my next pull request you may wonder if the _survive variation is needed given the _live variation, and the answer is yes - I had a crash that was reproducing only in the _survive case.)